src/main/java/uk/gov/digital/ho/hocs/casework/domain/model/StageWithCaseData.java

```diff
@@ -133,10 +133,13 @@ public StageWithCaseData(UUID caseUUID, String stageType, UUID teamUUID, UUID us
     }
 
     public void putData(String key, String value) {
+        if (this.data == null) {
+            this.data = new HashMap<>();
+        }
+
         var newData = new HashMap<>(this.data);
```

Comment:
Is this a little redundant? could we not just do `data.put` here and remove this?

Reply:
You wouldn't be able to do data.put if this.data was null, so it would need to be instantiated first

Reply:
Sorry I didn't explain there well. We keep the null check and then just add into the map instead of having a second map.